enclave/src/lib.rs

```diff
@@ -997,34 +1005,13 @@ where
 	Ok(Some((H256::from(call_hash), H256::from(operation_hash))))
 }
 
-pub fn get_onchain_storage_updates<O: EnclaveOnChainOCallApi>(
-	storage_hashes: Vec<Vec<u8>>,
-	header: &Header,
-	on_chain_ocall_api: &O,
-) -> Result<HashMap<Vec<u8>, Option<Vec<u8>>>> {
-	let requests = storage_hashes
-		.into_iter()
-		.map(|key| WorkerRequest::ChainStorage(key, Some(header.hash())))
-		.collect();
-
-	let update_map = on_chain_ocall_api
-		.worker_request::<Vec<u8>>(requests)
-		.map(into_storage_entry_iter)
-		.map(|i| verify_storage_entries(i, header))??;
-
-	Ok(update_map)
-}
-
-/// Verify a set of storage entries. This is defined here to keep the `substratee-storage` crate
-/// `no_std` compatible. Which it would not be with the `HashMap`.
-pub fn verify_storage_entries<Header: HeaderT>(
-	entries: impl Iterator<Item = StorageEntry<Vec<u8>>>,
-	header: &Header,
-) -> Result<HashMap<Vec<u8>, Option<Vec<u8>>>> {
-	let mut update_map = HashMap::new();
-	for e in entries {
-		e.verify_storage_proof(header)?;
-		update_map.insert(e.key, e.value);
+pub fn into_map(
+	storage_entries: Vec<StorageEntryVerified<Vec<u8>>>,
+) -> HashMap<Vec<u8>, Option<Vec<u8>>> {
+	let mut map = HashMap::new();
+	for e in storage_entries.into_iter() {
+		let ev = e.into_tuple();
+		map.insert(ev.0, ev.1);
 	}
-	Ok(update_map)
+	map
```

Comment:
Is there no nice stream/functional way of doing this? Like `collect` into a HashMap?

Reply:
Haha you are right, thanks for the hint!

https://stackoverflow.com/questions/30441698/how-do-i-create-a-map-from-a-list-in-a-functional-way/30441736